infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -387,26 +387,29 @@ private void trimPrefix() {
                                     + " attempting to trim anyways", sh.getSegment(),
                             sh.getRefCount());
                 }
-
                 sh.close();
+                writeChannels.remove(sh.getFileName());
+            }
+        }
 
-                File logFile = new File(sh.getFileName());
-                File trimmedFile = new File(getTrimmedFilePath(sh.getFileName()));
-                File pendingFile = new File(getPendingTrimsFilePath(sh.getFileName()));
+        File dir = new File(logDir);
+        FileFilter fileFilter = new FileFilter() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)